File: src/Creational_Patterns/Singleton/examples/java/singleton.java
public class SingletonClass {
    private static SingletonClass instance;
    private String data;

    private SingletonClass() {
        data = "Singleton data";
    }

    public static SingletonClass getInstance() {
        if (instance == null) {
            synchronized (SingletonClass.class) {
                if (instance == null) {
                    instance = new SingletonClass();
                }
            }
        }
        return instance;
    }

    public String getData() {
        return data;
    }
}

File: src/Creational_Patterns/Singleton/examples/ruby/singleton.rb
class SingletonClass
  attr_accessor :data

  def self.instance
    @instance ||= new
  end

  private_class_method :new

  private

  def initialize
    @data = "Singleton data"
  end
end

# Usage
singleton = SingletonClass.instance
puts singleton.data

# Attempt to create a new instance (will return the existing instance)
new_singleton = SingletonClass.new
puts new_singleton.data